### ISSUE TITLE
OOM reduction factor 1 for Tacotron2

### ISSUE THREAD

**Issue by huseinzol05** [TensorSpeech/TensorFlowTTS#427]
First of all, thank you very much for this repository, Tacotron2 implementation very easy to understand.

Before that, my organization not actually use your code directly to pretrain Tacotron 2, we have a standard to stick to Tensorflow 1.15, so we convert your Tacotron 2 code (do not worry, we put your license at the top) to make it able to run in Tensorflow 1.15 using https://github.com/tensorflow/tensorflow/tree/r1.15/tensorflow/contrib/seq2seq/python/ops. We use AdamW from official BERT, everything pretty look same including size of parameters, 31M. Max length mel is 950, max length text ids is 200, and batch size 32.

During the training, we hit OOM as we expected because `dynamic_decode` is storing each iterations, even when we tried batch size equal to 1 * Size of GPU, and we are using Tesla v100 32GB VRAM, unless if we set `swap_memory` equal to `True`, but the training session will be very slow, ~20s / step. After we set reduction factor to 3, everything able to run without swapping memory.

Based on your released models and other contributions, 32 batch size and reduction factor 1 seems like no problem to train without swapping memory. We read your code like insane to understand why it works, but not for us, is it because Tensorflow 2 handled gradient `while_loop` in `dynamic_decode` differently, or is it you have some implementation to make it able to run without swapping memory, we already hit our walls here.

Thank you very much.

**Comment by dathudeptrai** [comment]
> models and other contributions

I think it's because tf2, i saw many improvements of Tensorflow2 compared with tf1.

**Comment by huseinzol05** [comment]
I guess I will close this issue, thanks!
